### PR TITLE
Increase performance

### PR DESCRIPTION
**Pull request by WyriHaximus**
By ditching call_user_func_array and using argument unpacking (does bump minimum PHP version to 5.6)

### CODE REVIEW

File: src/Evenement/EventEmitterTrait.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function once($event, callable $listener)
         $onceListener = function () use (&$onceListener, $event, $listener) {
```
Comment (kelunik) [inline src/Evenement/EventEmitterTrait.php]: You can just use `...$args` then, too.

**Comment by bwoebi** [comment]
If you want to increase performance, just use `\call_user_func` with a leading backslash in order to have it optimized to native opcodes under PHP 7.

**Comment by WyriHaximus** [comment]
@bwoebi nice, care to make a PR so we can compare the difference? When the difference is small enough could bump this to `v3.0.0`.

File: src/Evenement/EventEmitterTrait.php (hunk 1 above)
Comment (WyriHaximus) [inline src/Evenement/EventEmitterTrait.php]: Just did and it barely seems to increase performance, does look better though

**Comment by WyriHaximus** [comment]
This question really at this point is to whether release this as `2.2` or `3.0` while bumping to PHP 7.0. This plus side of going with `2.2` is that there is no need to make [163 PR's](https://packagist.org/packages/evenement/evenement/dependents) while the public API doesn't change. This PR would be the only thing changing in `2.2` compared to `2.1` which will also include #42

**Comment by kelunik** [comment]
Why not both?

**Comment by WyriHaximus** [comment]
What ever gets in `2.1` gets in `3.0` :wink:

**Comment by PeeHaa** [comment]
If it breaks existing code it's a new major. I don't really see the question.  
It's fair of end users to expect things not to blow up between minor versions.

Also I don't see the need to make 163 PR's. Imo it's not your responsibility to update dependents. It's really up to the maintainers.

**Comment by WyriHaximus** [comment]
> If it breaks existing code it's a new major. I don't really see the question.
> It's fair of end users to expect things not to blow up between minor versions.

When making `v2.2` PHP 7+ only things won't blow up for current users because the BC break would be there when it would have been targeting PHP 5.6 as well. The thing is, is dropping PHP 5.6 support considered a BC break or not.

> Also I don't see the need to make 163 PR's. Imo it's not your responsibility to update dependants. It's really up to the maintainers.

Of course but when one package doesn't update you wouldn't get this improvement.

**Comment by PeeHaa** [comment]
> When making v2.2 PHP 7+ only things won't blow up for current users because the BC break would be there when it would have been targeting PHP 5.6 as well.

Yes obviously it's a BC break. I am running PHP 5.4 (which is valid using the current constraints) and bumping the version to anything above that is a breaking change and needs a major.

I am going to go as far as stating that when bumping the version using composer even if the code technically also works on 5.4 it still warrants a new major because it's a breaking change for me.

**Comment by kelunik** [comment]
> Yes obviously it's a BC break. I am running PHP 5.4 (which is valid using the current constraints) and bumping the version to anything above that is a breaking change and needs a major.

No, Composer just won't install the update for you. I don't consider it a BC break. It's often better than releasing a new major version if that's the only change.

**Comment by PeeHaa** [comment]
The point is that version 2.2 works for me. And 2.3 doesn't work anymore for me (at all).

How is that not a BC break?

**Comment by WyriHaximus** [comment]
> How is that not a BC break?

Because composer won't install it for you. Also I don't have a problem with making PR's to the 30 packages with the most downloads tbh.

**Comment by PeeHaa** [comment]
It doesn't matter whether it install it for you.

The minimum version is part of the public API through the composer config file.

Bumping the version is a BC break.

**Comment by PeeHaa** [comment]
I already gave my two cents where I explained why it's part of the public API.

Do whatever you want with the cents :P

**Comment by WyriHaximus** [comment]
> Do whatever you want with the cents :P

Much appreciated :+1: 

> I already gave my two cents where I explained why it's part of the public API.

This pretty much sells me on the `v3.0.0` for this PR :+1:

**Comment by WyriHaximus** [comment]
Thanks again for the feedback, much appreciated :+1:

**Pull request merged**